### ISSUE TITLE
Cannot open read-only files in bind-mounted volume Docker Desktop 2.2.0.4 on Windows

### ISSUE THREAD

**Issue by mrweiner** [lando/lando#2052]
Drupal projects which worked correctly on the previous RC throw permissions issues on `/app/<projectRoot>/sites/default/settings.php` as well as on images residing in `/app/<projectRoot>/sites/default/files`. This persisted after rebuilding and destroying the projects.

For anybody encountering this, going into the containers and executing `chmod -R 777 app/<projectRoot>/sites` does solve the problem, although it's obviously heavy-handed. There's probably a better way to deal with the issue.

**Comment by pirog** [comment]
f

**Comment by GoonerW** [comment]
This also occurs in Windows when the file is set readonly.

Using `attrib -r` on `settings.php` allowed the container to read the file.  Reversing it by running `attrib +r settings.php` recreates the issue.

Running ` attrib -r *.* /S` in the app root clears the read-only attribute on all files.

I think this is more due to Docker 2.2.0.4 as I ran DOcker 2.2.0.3 with Lando rc23 without any issues like this.  Upgrading to Docker 2.2.0.4 with Lando rc23 and this issue arose.

**Comment by Asterovim** [comment]
I have the same problem.
` chmod -R 777 app/<projectRoot>/sites `
have fixed my bug. Thanks you. I use windows 10 and lando lastest version (aft.1).

**Comment by mrweiner** [comment]
@GoonerW that makes sense for just settings.php, but doesn't fix the issue of perms on files/images/etc.

**Comment by GoonerW** [comment]
@mrweiner My original post had the command to remove the readonly attribute on all files in the app root.

I only used settings.php as an example to show what error is produced inside the container.

**Comment by mrweiner** [comment]
Got it, sorry I misunderstood.

**Comment by GoonerW** [comment]
No worries.  It's a weird issue.  Inside the container, settings.php has a permission of 555 but even 777 doesn't work.

**Comment by Asterovim** [comment]
Do you think fixed with lastest release (aft.2) ? is Related error : https://github.com/lando/lando/issues/2057 ??

**Comment by GoonerW** [comment]
@Asterovim No.  This "issue" occurs in the app container, not the database container.

I just checked anyway, same issue.  I "think" it's a Docker issue but I'm not sure.

**Comment by Asterovim** [comment]
@GoonerW Thanks you.

So if i delete my docker and i install docker 2.2.0.3 the problem will be fixed ?

**Comment by GoonerW** [comment]
@Asterovim Lando aft.1 and aft.2 require 2.2.0.4 but I guess it'll fix it.

**Comment by RichardDavies** [comment]
There's a docker patch that fixes this issue: https://github.com/docker/for-win/issues/6016#issuecomment-600285551

**Comment by mrweiner** [comment]
Looks like aft.2 + that Docker patch solved some other mysterious bugs I was experiencing. Thanks for the tip.

**Comment by pirog** [comment]
Just cut Lando 3.0.0-rrc1. which downgrades to Docker 2.2.0.3 to resolve this, def recommend everyone update to the latest Lando and then ensure their Docker version is 2.2.0.3

**Comment by JorisVanEijden** [comment]
Did you test this?
The installer doesn't downgrade anything:
```
2020-03-24 12:35:50.883   -- File entry --
2020-03-24 12:35:50.883   Dest filename: C:\Program Files\Lando\installers\docker\docker.exe
2020-03-24 12:35:50.883   Time stamp of our file: 2020-03-22 13:48:04.000
2020-03-24 12:35:50.883   Dest file exists.
2020-03-24 12:35:50.884   Time stamp of existing file: 2020-03-14 15:30:12.000
2020-03-24 12:35:50.884   Version of our file: 2.2.0.42716
2020-03-24 12:35:50.893   Version of existing file: 2.2.0.43472
2020-03-24 12:35:50.893   Existing file is a newer version. Skipping.
```

**Comment by pirog** [comment]
We test everything but we can’t fundamental change how someone else’s
installer or install packaging software works. If you have already upgraded
to 2.2.0.4 you need to manually downgrade.

**Comment by JorisVanEijden** [comment]
I'm sorry, I'm just frustrated by having to spend all day un and re-installing docker and lando.
I did not upgrade to 2.2.0.4 myself, lando did that in version "aft1".
The communication from lando was "Just cut Lando 3.0.0-rrc1. which downgrades to Docker 2.2.0.3 to resolve this" and "Downgraded Docker Desktop for Windows to 2.2.0.3 because #2052 #2076"

There was no mention of having to delete the docker installer from the program files directory if you have the lando aft1 release. Now there is, here in this ticket. You're welcome.

**Comment by pirog** [comment]
This is a free and open source project that is run by volunteers who spend thousands of uncompensated hours on top of their day job making software to make your life easier. We do the best we can and we 100% do not appreciate snide bullshit from users like you. 

If you don't like it, use something else i dont think anyone here will miss you

**Comment by JorisVanEijden** [comment]
I think you totally misread the tone of my message. I guess we're both frustrated by everything that's going on.

**Comment by pirog** [comment]
To anyone else that finds themselves here, please see the comment above https://github.com/lando/lando/issues/2052#issuecomment-602622236

Note the final part of that comment
> and then ensure their Docker version is 2.2.0.3

To be 100% explicit about what this means: if your Docker version _IS NOT_ `2.2.0.3` then you should _MANUALLY DOWNGRADE_ to that version because of a bug in Docker `2.2.0.4`. Here is the link
https://download.docker.com/win/stable/42716/Docker%20Desktop%20Installer.exe

**Comment by pirog** [comment]
@smutlord was nice enough to investigate this further and the deal is you _can_ setup the Lando installer so it will overwrite _newer_ installers with the`2.2.0.3` Docker installer. 

However, the Docker installer itself will not let you downgrade if it detects you have a newer version. This means that if you have already installed a version of Lando that installs Docker `2.2.0.4`, or have updated on your own accord, you will need to manually uninstall Docker and then rerun the Lando installer or manually install Docker `2.2.0.3`. Based on our metrics this will impact around 25 Lando users or in percentage terms 0.25% of users. 

We will kick out a new Lando version as soon as Docker rolls a fix but i think we've done about the best we can on the Lando side to help mitigate this.

**Comment by garrettw** [comment]
@pirog I was just installing rrc1 as a new installation on a mac, and the installer was bundled with Docker 2.2.0.4, not .3. Just FYI.

**Comment by pirog** [comment]
That is expected. rrc2 is the version with 2.2.0.3, but you need to note
the caveats in the comment above

**Comment by pirog** [comment]
Correction. Rrc1 is the version with 2.2.0.3 but you need to take note of
the caveat in the comment above, docker will not purposefully downgrade if
you already have 2.2.0.4

**Comment by garrettw** [comment]
> Also this ONLY applies to WINDOWS.
😄 so I just spent time manually installing 2.2.0.3 when I actually didn't need to. lol I should read better next time.

**Comment by mrweiner** [comment]
@pirog it sounds like this is fixed upstream with 2.2.0.5: https://github.com/docker/for-win/issues/6016#issuecomment-608014429
